### PR TITLE
fix: Allow `cluster_settings` to be list of maps instead of single map

### PR DESCRIPTION
**Pull request by ivan-sukhomlyn**
## Description
Define `cluster_settings` as a list of maps to possibly define other parameters in the future.

## Motivation and Context
Currently, a single **containerInsights** parameter can be defined.

## Breaking Changes
**No**, using flatten list at the `for_each` meta-argument allows to use both map and list value at the `cluster_settings` variable.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

define a custom value for `cluster_settings` as a list of maps or a map.
For example,
```HCL
  cluster_settings = [
    {
      name  = "containerInsights"
      value = "disabled"
    }
  ]
```
or
```HCL
  cluster_settings = {
    name  = "containerInsights"
    value = "disabled"
  }
```

```
$ pre-commit run -a
Terraform fmt............................................................Passed
Terraform wrapper with for_each in module................................Passed
Terraform validate.......................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
```

### CODE REVIEW

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
why is this change warranted if the only value that ECS supports is containerInsights? https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ecs_cluster#setting

**Comment by ivan-sukhomlyn** [comment]
@bryantbiggs maybe it's a wrong declaration of a PR from my side and needs to be renamed to the `refactor!:`
Mainly, the change is for possible further extension of settings (for example, adding a new setting) for the ECS cluster and due to the setting actually being dynamically defined via `for_each` wrapping a map into a set.

**Comment by bryantbiggs** [comment]
@antonbabenko what are your thoughts on creating a temporary v6 branch to pull in these various PRs that are breaking changes and then once those are complete, merge in that branch as the next major release? Trying to think of a way to allow crowd sourcing these changes instead of the normal way (i.e. - one person bunches everything up in a breaking change PR on their own)? 

I started a milestone to track these changes https://github.com/terraform-aws-modules/terraform-aws-ecs/milestone/1

File: modules/cluster/main.tf
Comment (antonbabenko) [inline modules/cluster/main.tf:82]:
```suggestion
    for_each = flatten([var.cluster_settings])
```

This will be equal to `list(map(string))` for both types: `list(map(string))` and `map(string)`.

File: modules/cluster/variables.tf
Comment (antonbabenko) [inline modules/cluster/variables.tf:31]:
```suggestion
  type        = any
```

`any` is more broad and not introducing BC break.

File: variables.tf
Comment (antonbabenko) [inline variables.tf:31]:
```suggestion
  type        = any
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]
I think we can make it without introducing the BC break.

@bryantbiggs We can have `v6` branch for small crowd-sourcing PRs like you said, but I don't think that this one is actually breaking anything and we can make it to `v5` without introducing breaking changes.

File: modules/cluster/main.tf
Comment (ivan-sukhomlyn) [reply modules/cluster/main.tf:82]: @antonbabenko thanks for suggestions!
I'll prepare the appropriate changes.

**Comment by ivan-sukhomlyn** [comment]
@bryantbiggs I've tested with `flatten` list at the `for_each` meta-argument with **`cluster_settings`** variable value.
In all cases, there are no errors and changes of resources with both types of the `cluster_setting` variable.
```HCL
  cluster_settings = [{
    name  = "containerInsights"
    value = "disabled"
  }]
```
or
```
  cluster_settings = {
    name  = "containerInsights"
    value = "disabled"
  }
```
The PR description has been updated accordingly.

@antonbabenko thanks for the suggestion 💪

**Comment by antonbabenko** [comment]
@ivan-sukhomlyn Looks good to me, and it is not a breaking change anymore. :)

@bryantbiggs I don't know what has changed in GitHub, but we suddenly get "WIP" checks blocking the merge. I will disable it as soon as I see it.

**Review by antonbabenko — APPROVED** [review]

**Review by antonbabenko — APPROVED** [review]

**Review by bryantbiggs — APPROVED** [review]
its the WIP bot that just won't die 😅

**Comment by antonbabenko** [comment]
This WIP bot made everybody look busy working on everything at once :)

**Comment by antonbabenko** [comment]
This PR is included in version 5.8.1 :tada:

**Pull request merged**
